src/api/procedures/addInstruction.ts

```diff
@@ -114,6 +115,43 @@ export const createAddInstructionResolver =
     return result;
   };
 
+/**
+ * @hidden
+ */
+function getEndCondition(
+  instruction: AddInstructionParams,
+  latestBlock: BigNumber,
+  index: number
+): { endCondition: InstructionEndCondition; errorIndex: number | null } {
+  let endCondition: InstructionEndCondition;
+  let errorIndex = null;
+
+  if ('endBlock' in instruction && instruction.endBlock) {
```

Comment:
<img width="1284" alt="Screenshot 2023-02-01 at 7 08 48 PM" src="https://user-images.githubusercontent.com/34747455/216058492-0e94bc58-08ff-40b8-8985-d2436dd9d877.png">

The interface has optional fields based on the type that's why the 'in' is needed for compiler understanding and then to make sure there is some value in endBlock the extra condition. Is there something else I could do here maybe with structure of interface?

Reply:
I see, TS won't allow you to optionally access that variable since it might not be defined, even though the JS isn't an error.

OK to leave. I think the only other option is to use a "type" field, but thats not worth it here.